### ISSUE TITLE
Allow access to EventFrames

### ISSUE THREAD

**Issue by Hugovdberg** [Hugovdberg/PIconnect#493]
Feature request
===============

Abstract
--------

Export the `EventFrame` functionality in a manner similar to `PIAFElements`

Motivation and summary
----------------------

The `EventFrame` functionality of PI AF is currently only available through the raw SDK methods,
and not through the PIconnect classes.

Suggested solution
------------------

Add a new class `PIAFEventFrame`.

Rejected options
----------------

NA

Additional context
------------------

NA

**Comment by Hugovdberg** [comment]
A snippet to get the data from event frames generated by analyses on `AFAttribute`s:

```python
import PIconnect as PI


element_path = r"Root Element\Child element"
event_frame_start_time = "*-365d"

database = PI.PIAFDatabase()
element = database.descendant(element_path)
analyses = [
    e
    for e in element.element.Analyses
    if e.Name == "Analysis name"
]

eventframes = []
for analysis in analyses:
    eventframes.extend(
        PI.AFSDK.AF.EventFrame.AFEventFrame.FindEventFramesByAnalysis(
            analysis,
            PI.AFSDK.AF.Time.AFTime.Parse(event_frame_start_time),
            0,
            12000,
            PI.AFSDK.AF.EventFrame.AFEventFrameSearchMode.ForwardFromStartTime,
            "",
            "",
            None,
            None,
            True,
        )
    )

for frame in eventframes:
    print(f"------------------\nFrame name: {frame.Name}\n------------------")
    for attribute in frame.Attributes:
        attr = PI.PIAF.PIAFAttribute(element=frame, attribute=attribute)
        values = attr.recorded_values(
            frame.StartTime, frame.EndTime, "inside"
        ).drop_duplicates()
        print(values)
        print()
    print()
```

**Comment by Hugovdberg** [comment]
Currently there are some issues with extracting event frames, mainly because the API has changed on a few occasions. I'm testing against API versions 2.7 and 2.9, but I can't test against API version 2.10. Some of the older methods have become obsolete in version 2.10, and some of the newer methods aren't available in 2.7. I'm contemplating on how to make this work across different API versions.

**Comment by freitaass** [comment]
Is this feature still available? are there any support for extract attribute values in eventframes?
